scipy/optimize/tests/test_linprog.py

```diff
@@ -1172,6 +1172,10 @@ def test_optimize_result(self):
         assert_allclose(c @ res.x, res.fun)
         assert_allclose(b_eq - A_eq @ res.x, res.con, atol=1e-11)
         assert_allclose(b_ub - A_ub @ res.x, res.slack, atol=1e-11)
+        for key in ['eqlin', 'ineqlin', 'lower', 'upper']:
+            if key in res.keys():
+                assert isinstance(res[key]['marginals'], np.ndarray)
+                assert isinstance(res[key]['residual'], np.ndarray)
```

Comment:
This indeed looks like what Matt was asking for--I'll let them confirm.

Reply:
Yes, I was just going to test it locally and then merge. @mckib2 look good to you, too?

Reply:
Yep, merge away!